pkg/loop/datasource.go

```diff
@@ -0,0 +1,53 @@
+package loop
+
+import (
+	"context"
+	"math/big"
+
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+
+	pb "github.com/smartcontractkit/chainlink-relay/pkg/loop/internal/pb"
+)
+
+var _ median.DataSource = (*dataSourceClient)(nil)
+
+type dataSourceClient struct {
+	grpc pb.DataSourceClient
+}
+
+func newDataSourceClient(cc *grpc.ClientConn) *dataSourceClient {
+	return &dataSourceClient{grpc: pb.NewDataSourceClient(cc)}
+}
+
+func (d *dataSourceClient) Observe(ctx context.Context) (*big.Int, error) {
+	reply, err := d.grpc.Observe(ctx, &emptypb.Empty{})
+	if err != nil {
+		return nil, err
+	}
+	i := new(big.Int)
+	i.SetBytes(reply.Value)
```

Comment:
should there be error handling or validation here? i'm surprised it's ok to set this to any byte slice

Reply:
Nah, it just returns itself: https://pkg.go.dev/math/big#Int.SetBytes
I suppose there are no invalid values since it is just interpreted as a big-endian, but now I see that this is actually a lossy conversion that does absolute-value and then assumes uint :thinking: We'll have to fix this somehow - I don't see a drop-in google.golang.org/protobuf